preshed/maps.pyx

```diff
@@ -218,48 +205,44 @@ cdef bint map_iter(const MapStruct* map_, int* i, key_t* key, void** value) nogi
 
 
 @cython.cdivision
-cdef inline Cell* _find_cell(Cell* cells, const key_t size, const key_t key) nogil:
+cdef inline Cell _find_cell(const vector[Cell]& cells, const key_t key) nogil:
     # Modulo for powers-of-two via bitwise &
-    cdef key_t i = (key & (size - 1))
+    cdef key_t i = (key & (cells.size() - 1))
     while cells[i].key != EMPTY_KEY and cells[i].key != key:
-        i = (i + 1) & (size - 1)
-    return &cells[i]
+        i = (i + 1) & (cells.size() - 1)
+    return cells[i]
 
 
 @cython.cdivision
-cdef inline Cell* _find_cell_for_insertion(Cell* cells, const key_t size, const key_t key) nogil:
+cdef inline vector[Cell].iterator _find_cell_for_insertion(vector[Cell]* cells, const key_t key) nogil:
```

Comment:
Any specific reason why we're using a pointer to the `vector[Cell]` instead of a reference? We should be able to get rid off the `deref` calls if we use the latter, no?

Reply:
Yes. To make it explicit to the call side that you are giving the object for modification. Contrast with `const &`, where it's irrelevant to the call site if the object is passed by value or reference, since it's not mutated.

http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f17-for-in-out-parameters-pass-by-reference-to-non-const
https://google.github.io/styleguide/cppguide.html#Inputs_and_Outputs